collector/sys_user_summary.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 Eduardo J. Ortega U.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const sysUserSummaryQuery = `
+	SELECT
+		user,
+		statements,
+		statement_latency,
+		statement_avg_latency,
+		table_scans,
+		file_ios,
+		file_io_latency,
+		current_connections,
+		total_connections,
+		unique_hosts,
+		current_memory,
+		total_memory_allocated
+	FROM
+		` + sysSchema + `.x$user_summary
+`
+
+var (
+	sysUserSummaryStatements = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "statements"),
+		" The total number of statements for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryStatementLatency = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "statement_latency"),
```

Comment:
Do we know what unit this is? I checked the MySQL docs and it doesn't say.

Whatever this is, we should convert it to float64 seconds.

And similar to all the other counters, it should end in `_total`.

Reply:
```suggestion
		prometheus.BuildFQName(namespace, sysSchema, "statement_seconds_total"),
```